megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -1948,30 +1948,27 @@ FiringPlan getBestFiringPlanWithTwists(Entity shooter, Targetable target,
             return noTwistPlan;
         }
 
-        // Turn to the right.
-        shooter.setSecondaryFacing(correctFacing(originalFacing + 1));
-        FiringPlan rightTwistPlan = getBestFiringPlan(shooter, target, game, ammoConservation);
-        rightTwistPlan.setTwist(1);
-
-        // Turn to the left.
-        shooter.setSecondaryFacing(correctFacing(originalFacing - 1));
-        FiringPlan leftTwistPlan = getBestFiringPlan(shooter, target, game, ammoConservation);
-        leftTwistPlan.setTwist(-1);
-
-        // todo extended torso twist.
-
+        ArrayList<Integer> validFacingChanges = getValidFacingChanges(shooter);
```

Comment:
Typically, it's better to use the interface (List in this case) for variable declarations as it reduces overhead and makes it easier to convert to other List subtypes if needed.